### ISSUE TITLE
failed to create windows under wayland

### ISSUE THREAD

**Issue by fthalabi** [marcusva/py-sdl2#162]
Hi, I'm using the following script for testing
```
import sys
import sdl2
import sdl2.ext


def main():
    sdl2.ext.init()
    window = sdl2.ext.Window("PySDL2 Test", size=(320, 200))
    window.show()

    window.get_surface()

    sdl2.ext.TestEventProcessor().run(window)

    sdl2.ext.quit()
    return 0


if __name__ == "__main__":
    sys.exit(main())
```
the code works with the x11 driver
`SDL_VIDEODRIVER=x11 python test.py `
but it doesn't with the wayland one.
`SDL_VIDEODRIVER=wayland python test.py`
```
Traceback (most recent call last):
  File "test.py", line 20, in <module>
    sys.exit(main())
  File "test.py", line 8, in main
    window = sdl2.ext.Window("PySDL2 Test", size=(320, 200))
  File "/home/fenho/.local/lib64/python3.6/site-packages/sdl2/ext/window.py", line 49, in __init__
    raise SDLError()
sdl2.ext.common.SDLError: b'failed to create a window surface'
```
i tested the change proposed in #139 to sdl2/ext/window.py with no effect. 

Also, to verify that sdl is working correctly i use with the following. 
```
#include <SDL2/SDL.h>
#include <iostream>

int main()
{
    if(SDL_Init(SDL_INIT_VIDEO) < 0)
    {
        std::cout << "Failed to initialize the SDL2 library\n";
        return -1;
    }

    std::cout << "SDL_GetCurrentVideoDriver: " << SDL_GetCurrentVideoDriver() << "\n";

    SDL_Window *window = SDL_CreateWindow("SDL2 Window",
                                          SDL_WINDOWPOS_CENTERED,
                                          SDL_WINDOWPOS_CENTERED,
                                          680, 480,
                                          0);

    if(!window)
    {
       	std::cout << "Failed to create window\n";
       	return -1;
    }
    SDL_Surface *window_surface = SDL_GetWindowSurface(window);

    if(!window_surface)
    {
        std::cout << "Failed to get the surface from the window\n";
        return -1;
    }

    SDL_UpdateWindowSurface(window);

    SDL_Delay(5000);
}
```
compiled with: `g++ -o test3.o test3.cpp  $(sdl2-config --cflags --libs)`
output: 
```
SDL_GetCurrentVideoDriver: wayland
## shows black window
```

**Comment by a-hurst** [comment]
This is a weird bug, thanks for bringing it to our attention! It looks different than the reported in #139 since you're getting an actual SDL error on window creation instead of just failing to show a successfully created window.

Can you report what version of SDL2 you're using? If you update to the latest PyPI version of pysdl2 (0.9.7), you can get the version pysdl2 is loading via `sdl2.dll.version`. I also dug into the SDL source based on the error text and it looks like throws that specific error when it fails to create an EGL surface for the Wayland window, which *seems* like something far outside of pysdl2's wheelhouse, but your C code seems to suggest otherwise.

Also, could you try the following code on your end and see what you get?

```python
import sdl2

sdl2.SDL_Init(sdl2.SDL_INIT_VIDEO)

vid_driver = sdl2.SDL_GetCurrentVideoDriver().decode('utf-8')
print("Current video driver: {0}".format(vid_driver))

window = sdl2.SDL_CreateWindow(
    "SDL2 Window",
    sdl2.SDL_WINDOWPOS_CENTERED, sdl2.SDL_WINDOWPOS_CENTERED,
    680, 480, 0
)
if not window:
    raise sdl2.SDLError()

sdl2.SDL_ShowWindow(window)
```

**Comment by fthalabi** [comment]
Sorry for the delay, here is the package version.

`pip show PySDL2`
```
Name: PySDL2
Version: 0.9.7
Summary: Python SDL2 bindings
Home-page: https://github.com/marcusva/py-sdl2
Author: Marcus von Appen
Author-email: marcus@sysfault.org
License: Public Domain / zlib
Location: /home/fenho/backup/dotfiles/.local/lib64/python3.6/site-packages
Requires:
Required-by:
```
and the requested code.
```
Python 3.6.9 (default, Feb  9 2020, 23:34:11)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.1.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import sdl2

In [2]: sdl2.dll.version
Out[2]: 2010
In [3]: %paste

   ...: sdl2.SDL_Init(sdl2.SDL_INIT_VIDEO)
   ...:
   ...: vid_driver = sdl2.SDL_GetCurrentVideoDriver().decode('utf-8')
   ...: print("Current video driver: {0}".format(vid_driver))
   ...:
   ...: window = sdl2.SDL_CreateWindow(
   ...:     b"SDL2 Window",
   ...:     sdl2.SDL_WINDOWPOS_CENTERED, sdl2.SDL_WINDOWPOS_CENTERED,
   ...:     680, 480, 0
   ...: )
   ...: if not window:
   ...:     print(sdl2.SDL_GetError())
   ...:
   ...: sdl2.SDL_ShowWindow(window)
Current video driver: wayland
b'failed to create a window surface'

```
thanks for the help.

**Comment by a-hurst** [comment]
Just set up an old laptop with SwayWM to test this, and with the latest SDL2 binaries I can't reproduce the issue using the same code. Closing this for now, assuming it's a configuration-specific issue or something fixed in SDL 2.0.14.
